sites/game-client/ui/overlay.mjs

```diff
@@ -61,8 +76,11 @@ class Overlay {
   */
   onClickToolbarButton() {
     // WIP(fanlan1210)
-    if (this.canHide() && this.position !== undefined) this.hide();
-    else this.show();
+    if (this.canHide() && this.position !== undefined) {
+      this.positionCache = this.position; // store current position state
+      this.hide();
+    }
+    else this.show(this.positionCache); // resore position state
```

Comment:
We might need a default for this.positionCache, or maybe you can add a getDefaultPosition() to Overlay that we'll require the derived class to implement?